plugins/commands/compare/src/main/kotlin/CompareCommand.kt

```diff
@@ -158,12 +169,22 @@ private class InvariantInstantSerializer : StdSerializer<Instant>(Instant::class
     }
 }
 
+private class EpochInstantDeserializer : StdDeserializer<Instant>(Instant::class.java) {
+    override fun deserialize(parser: JsonParser, context: DeserializationContext): Instant =
+        Instant.EPOCH.also { parser.codec.readTree<JsonNode>(parser) }
```

Comment:
Doesn't this need to say `Instant` instead of `JsonNode`, so the parser is advanced for the right amount of properties for the respective type?

Reply:
`readTree` works only for `JsonNode` ([docs](https://javadoc.io/static/com.fasterxml.jackson.core/jackson-core/2.9.9/com/fasterxml/jackson/core/ObjectCodec.html#readTree-com.fasterxml.jackson.core.JsonParser-))

Reply:
Right, you'd obviously need to use `readValue` then.

Reply:
But I think I have to use `readTree` at this point, otherwise the parser would point to the same place after it was deserialized and this causes an exception.

Reply:
Ah, yes, you'd probably run into a recursive problem here.

But how does `readTree` know what the structure of an `Instant` is, and how many properties to read / skip? I guess this just happens to work because `Instant` is serialized as a string which happens to a single primitive `JsonNode` / `TextNode`.

Anyway, I believe you've verified this to work.

Reply:
> read the "object" the parser is currently pointing at no matter if it is a simple value or a nested tree

You're right, it will read the whole next node, no matter whether it's a primitive text node or a complex object node.